pkg/cluster/cluster.go

```diff
@@ -433,28 +441,34 @@ func (c *Cluster) pollPods() (running, pending []*v1.Pod, err error) {
 			c.logger.Warningf("ignoring pod %q with unexpected owner %q", pod.Name, pod.OwnerReferences[0].UID)
 			continue
 		}
-		// Compute the list of running and pending pods.
-		// TODO: Take into account failed pods, so they can be handled as well, or use an adequate restart policy when creating pods.
+		// Add the current pod to the appropriate slice based on the current phase.
 		switch pod.Status.Phase {
 		case v1.PodRunning:
 			running = append(running, pod)
 		case v1.PodPending:
-			pending = append(pending, pod)
-		default:
-			c.logger.Warningf("ignoring pod %q with unexpected phase %q", pod.Name, pod.Status.Phase)
+			fallthrough
+		case v1.PodUnknown:
+			waiting = append(waiting, pod)
+		case v1.PodSucceeded:
```

Comment:
ok this would mean that after lame duck exit 0 for example the operator would replace the pod as well

Reply:
No, it just deletes said pod. Only if the cluster size is greater than the number of pods, will a new one be provisioned.

Reply:
can we punt deleting when COMPLETED/Succeeded for now? this means that pods that are terminated with `kubectl exec <pod> /gnatsd -sl=ldm=<pid>` would be replaced too by the operator, if so how can we delete the cluster pods with a graceful exit?

Reply:
My suggestion is enable LDM and scale the cluster to 0 before executing LDM.